core/src/main/java/org/infinispan/manager/DefaultCacheManager.java

```diff
@@ -1019,10 +1023,28 @@ public ClusterExecutor executor() {
       JGroupsTransport transport = (JGroupsTransport) globalComponentRegistry.getComponent(Transport.class);
       if (transport != null) {
          long time = getCacheManagerConfiguration().transport().distributedSyncTimeout();
-         return new ClusterExecutorImpl(null, this, transport, time, TimeUnit.MILLISECONDS,
-                 globalComponentRegistry.getComponent(ExecutorService.class, KnownComponentNames.REMOTE_COMMAND_EXECUTOR));
+         return ClusterExecutors.allSubmissionExcecutor(null, this, transport, time, TimeUnit.MILLISECONDS,
+               globalComponentRegistry.getComponent(ExecutorService.class, KnownComponentNames.REMOTE_COMMAND_EXECUTOR),
+               globalComponentRegistry.getComponent(ScheduledExecutorService.class, KnownComponentNames.TIMEOUT_SCHEDULE_EXECUTOR));
       } else {
-         return new ClusterExecutorImpl(null, this, null, -1, TimeUnit.MILLISECONDS, ForkJoinPool.commonPool());
+         ScheduledExecutorService timeoutExecutor = globalComponentRegistry.getComponent(ScheduledExecutorService.class,
+               KnownComponentNames.TIMEOUT_SCHEDULE_EXECUTOR);
+         if (timeoutExecutor == null) {
+            Hack hack = new Hack();
+            globalComponentRegistry.wireDependencies(hack);
+            timeoutExecutor = hack.timeoutExecutor;
+         }
+         return ClusterExecutors.allSubmissionExcecutor(null, this, null,
+               TransportConfiguration.DISTRIBUTED_SYNC_TIMEOUT.getDefaultValue(), TimeUnit.MILLISECONDS, ForkJoinPool.commonPool(),
+               timeoutExecutor);
+      }
+   }
+
+   static class Hack {
```

Comment:
Going to try to add `ClusterExecutor` global component factory instead which can inject it.